index.js

```diff
@@ -97,11 +99,34 @@ module.exports = {
     'import/order': [
       'error',
       {
-        groups: ['builtin', 'external', 'parent', 'sibling', 'index'],
+        groups: [
+          'builtin',
+          'external',
+          'parent',
+          'sibling',
+          'index',
+          'object',
+          'type',
+        ],
         'newlines-between': 'always',
       },
     ],
 
+    // Ensure consistent use of file extension within the import path
+    // The airbnb config we extend does not support TypeScript so we override it here
+    // https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md
+    'import/extensions': [
+      'error',
+      'ignorePackages',
+      {
+        js: 'never',
+        jsx: 'never',
+        mjs: 'never',
+        ts: 'never',
+        tsx: 'never',
+      },
```

Comment:
We use the airbnb config here as our base: https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/imports.js#L140-L144

We need to extend it so that TypeScript projects can resolve `.ts?(x)` extensions.